test/common-tests/acbs-get-facility-by-identifier-api-tests.ts

```diff
@@ -0,0 +1,79 @@
+import { TIME_EXCEEDING_ACBS_TIMEOUT } from '@ukef-test/support/environment-variables';
+import nock from 'nock/types';
+import supertest from 'supertest';
+
+export const withAcbsGetFacilityServiceCommonTests = ({
+  givenTheRequestWouldOtherwiseSucceed,
+  requestToGetFacility,
+  makeRequest,
+}: {
+  givenTheRequestWouldOtherwiseSucceed: () => void;
+  requestToGetFacility: () => nock.Interceptor;
+  makeRequest: () => supertest.Test;
+}) => {
+  describe('Common ACBS get facility service tests', () => {
+    it('returns a 404 response if ACBS get facility returns a 400 response with the string "The facility not found"', async () => {
+      givenTheRequestWouldOtherwiseSucceed();
+      requestToGetFacility().reply(400, 'The facility not found');
+
+      const { status, body } = await makeRequest();
+
+      expect(status).toBe(404);
+      expect(body).toStrictEqual({
+        statusCode: 404,
+        message: 'Not found',
+      });
+    });
+
+    it('returns a 500 response if ACBS get facility returns a 400 response without the string "The facility not found"', async () => {
+      givenTheRequestWouldOtherwiseSucceed();
+      requestToGetFacility().reply(400, 'An error message from ACBS.');
+
+      const { status, body } = await makeRequest();
+
+      expect(status).toBe(500);
+      expect(body).toStrictEqual({
+        statusCode: 500,
+        message: 'Internal server error',
+      });
+    });
+
+    it('returns a 500 response if ACBS get facility responds with a 400 response that is not a string', async () => {
+      givenTheRequestWouldOtherwiseSucceed();
+
+      const acbsErrorMessage = { Message: 'error message' };
+      requestToGetFacility().reply(400, acbsErrorMessage);
+
+      const { status, body } = await makeRequest();
+
+      expect(status).toBe(500);
+      expect(body).toStrictEqual({ message: 'Internal server error', statusCode: 500 });
+    });
+
+    it('returns a 500 response if ACBS get facility responds with a status code that is NOT 200 or 400', async () => {
```

Comment:
We've been returning a 500 for unexpected errors from ACBS. E.g. if ACBS returns a 401 to us then that means something has gone wrong in our system because we should always be sending authenticated requests, i.e. it's a 500 for _our_ client, not a 401